### PR TITLE
fix(flat-web): support range in service-worker

### PR DESCRIPTION
**Pull request by hyrious**

### CODE REVIEW

File: web/flat-web/public/worker.js [hunk 1]
```diff
@@ -20,6 +20,37 @@ self.onfetch = event => {
         event.respondWith(
             pptConvertCache.then(async cache => {
                 const response = await cache.match(request);
+
+                // https://web.dev/sw-range-requests
+                const range = request.headers.get("range");
+                if (range && response) {
+                    if (response.status === 206) return response;
```
Comment (BlackHole1) [inline web/flat-web/public/worker.js]:
```suggestion
                    if (response.status === 206) {
                        return response;
                    }
```

File: web/flat-web/public/worker.js [hunk 2]
```diff
@@ -20,6 +20,37 @@ self.onfetch = event => {
         event.respondWith(
             pptConvertCache.then(async cache => {
                 const response = await cache.match(request);
+
+                // https://web.dev/sw-range-requests
+                const range = request.headers.get("range");
```
Comment (BlackHole1) [inline web/flat-web/public/worker.js:25]:
```suggestion
                const hasRange = request.headers.get("range");
```

File: web/flat-web/public/worker.js [hunk 3]
```diff
@@ -20,6 +20,37 @@ self.onfetch = event => {
...
+                        slicedResponse.headers.set(
+                            "Content-Range",
+                            `bytes ${start}-${end}/${blob.size}`,
+                        );
+                        return slicedResponse;
+                    } catch (error) {
+                        return new Response("", {
+                            status: 416,
+                            statusText: "Range Not Satisfiable",
+                        });
```
Comment (BlackHole1) [inline web/flat-web/public/worker.js]:
```suggestion
                        console.error(error);
                        return fetch(request);
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: web/flat-web/public/worker.js (hunk 2 above)
Comment (hyrious) [reply web/flat-web/public/worker.js:25]: `range` is used below, not a boolean.

Comment (BlackHole1) [reply web/flat-web/public/worker.js:25]: ok

**Review by BlackHole1 — APPROVED** [review]

**Pull request merged**